query/resource_filter.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2022-2023 Searis AS
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package query
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// Filter describe a type can return an internal resource filter structure.
+// Filters reduce which resources.
+type Filter interface {
+	filter() filter
+}
```

Comment:
Public interface with private seems to be awkward, maybe better to make it private?